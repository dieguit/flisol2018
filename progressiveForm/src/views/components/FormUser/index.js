import React from 'react';
import PropTypes from 'prop-types';
import { Form, Segment, Message } from 'semantic-ui-react';
import { Field, Confirm } from '../../utils/ProgressiveForm';

const FormUser = ({ stepError, stepValues: { username } }) => {
  const isConfirmDisabled = () => stepError.length > 0 || !username || username.length === 0;
  return (
    <Segment>
      <h2>Step 3</h2>
      <Message hidden={stepError.length === 0} content={stepError} negative />
      <Field name="username" component={(props) => <Form.Input label="User" {...props} />} />
      <Confirm
        component={(props) => (
          <Form.Button primary disabled={isConfirmDisabled()} content="Confirm" {...props} />
        )}
      />
    </Segment>
  );
};

FormUser.defaultProps = {
  stepError: '',
  stepValues: {},
};

FormUser.propTypes = {
  stepError: PropTypes.string,
  stepValues: PropTypes.shape({
    username: PropTypes.string,
  }),
};

export default FormUser;
