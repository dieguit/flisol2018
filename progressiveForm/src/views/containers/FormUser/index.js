import React from 'react';
import { progressiveFormStep } from '../../utils/ProgressiveForm';
import FormUser from '../../components/FormUser';
import { validate } from '../../utils/api';

const FormUserContainer = (props) => <FormUser {...props} />;

const validateAsync = (values) => {
  return validate(values.username);
};

const formStep = progressiveFormStep({
  formId: 'progForm1',
  stepId: 'username',
  stepsRequired: ['a', 'b'],
  validate: validateAsync,
  confirm: true,
})(FormUserContainer);

export default formStep;
