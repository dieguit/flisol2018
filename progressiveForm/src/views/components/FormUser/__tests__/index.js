import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import FormUser from '../index';

jest.mock('../../../utils/ProgressiveForm/progressiveFormField', () => 'MockField');
jest.mock('../../../utils/ProgressiveForm/progressiveFormConfirm', () => 'MockConfirm');

function setup() {
  const props = {
    stepError: 'This is an error',
    stepValues: { username: 'User007' },
  };
  const wrapper = shallow(<FormUser {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('FormUser Component', () => {
  it('should match snapshot', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<FormUser />);
    expect(result).toMatchSnapshot();
  });

  it('should render self and subcomponents', () => {
    const { wrapper } = setup();
    expect(wrapper.find('h2').text()).toEqual('Step 3');

    const MessageProps = wrapper.find('Message').props();
    expect(MessageProps.hidden).toBe(false);
    expect(MessageProps.content).toEqual('This is an error');

    expect(wrapper.find('MockField').exists()).toBe(true);
    expect(wrapper.find('MockConfirm').exists()).toBe(true);
  });
});

jest.unmock('../../../utils/ProgressiveForm/progressiveFormField', () => 'MockField');
jest.unmock('../../../utils/ProgressiveForm/progressiveFormConfirm', () => 'MockConfirm');
