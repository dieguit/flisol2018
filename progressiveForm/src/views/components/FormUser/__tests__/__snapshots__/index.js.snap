// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormUser Component should match snapshot 1`] = `
<Segment>
  <h2>
    Step 3
  </h2>
  <Message
    content=""
    hidden={true}
    negative={true}
  />
  <MockField
    component={[Function]}
    name="username"
  />
  <MockConfirm
    component={[Function]}
  />
</Segment>
`;
